### PR TITLE
add shopify integration

### PR DESCRIPTION
**Pull request by jrhizor**
Adds Shopify integration. I need to configure credentials for standard tests to pass but everything else should be there.

Most unique content is in the source python file itself (the check code) and `shopify.md`.

### CODE REVIEW

File: airbyte-integrations/connectors/source-shopify-singer/sample_files/catalog.json [hunk 1]
```diff
@@ -0,0 +1,3702 @@
+{
```
Comment (jrhizor) [inline airbyte-integrations/connectors/source-shopify-singer/sample_files/catalog.json:1]: this is not the full catalog, only streams that don't have pii

File: airbyte-integrations/connectors/source-shopify-singer/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+development environment of choice. To activate the venv from the terminal, run:
+```
+cd airbyte-integrations/connectors/source-shopify # cd into the connector directory
+source .venv/bin/activate
+```
+If you are in an IDE, follow your IDE's instructions to activate the virtualenv.
+
+#### Create credentials
+**If you are a community contributor**, follow the instructions in the [documentation](https://docs.airbyte.io/integrations/sources/shopify)
+to generate the necessary credentials. Then create a file `secrets/config.json` conforming to the `source_shopify_singer/spec.json` file.
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-shopify-singer/README.md:26]: is it relatively straight foward for a user to find all of these creds themselves or are there any docs to link to help them?

File: airbyte-integrations/connectors/source-shopify-singer/setup.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    install_requires=["airbyte-protocol", "tap-shopify==1.2.6", "ShopifyAPI==8.0.4"],
+    package_data={"": ["*.json"]},
+    setup_requires=["pytest-runner"],
+    tests_require=["pytest"],
+    extras_require={
+        # Dependencies required by the main package but not integration tests should go in main. Deps required by
+        # integration tests but not the main package go in integration_tests. Deps required by both should go in
+        # install_requires.
+        "main": ["base-singer", "base-python"],
+        "tests": ["airbyte_python_test", "pytest"],
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-shopify-singer/setup.py:42]:
```suggestion
        "tests": ["airbyte-python-test", "pytest"],
```

we need to fix this in the template.

File: airbyte-integrations/connectors/source-shopify-singer/source_shopify_singer/source.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class SourceShopifySinger(SingerSource):
+    def __init__(self):
+        super().__init__()
+
+    def transform_config(self, raw_config):
+        return {"start_date": raw_config["start_date"], "api_key": raw_config["api_password"], "shop": raw_config["shop"]}
+
+    def check(self, logger: AirbyteLogger, config_container: ConfigContainer) -> AirbyteConnectionStatus:
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-shopify-singer/source_shopify_singer/source.py:40]: nice. this is probably the better way of doing this (using the integration's sdk) relative to some of the hack and slash stuff i've done.

**Review by cgardens — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/source-shopify-singer/README.md (hunk 2 above)
Comment (jrhizor) [reply airbyte-integrations/connectors/source-shopify-singer/README.md:26]: This is non-trivial but described in `shopify.md`

File: airbyte-integrations/connectors/source-shopify-singer/sample_files/config.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "start_date": "2020-01-30",
+  "api_password": "<Shopify API Password>",
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-shopify-singer/sample_files/config.json:3]: re a comment in another PR: is it the name Shopify gives it: api_password?

**Pull request merged**